### ISSUE TITLE
hlsearch+colors+syntax highlighting bug

### ISSUE THREAD

**Issue by AbePralle** [macvim-dev/macvim#1291]
### Steps to reproduce

1. Create `.vimrc` containing only these lines:
```
set hlsearch
colors darkblue "or any colorscheme
syntax on       "or 'syntax enable'
```
2. Edit any file (`mvim filename`), create a duplicate line or two, then search for a duplicate word.
3. All matches will be highlighted except the match at the current cursor position (press `n` or `*` to cycle through and observe).

- Note: changing the color scheme OR re-issuing `syntax on` after MacVim launches will fix the issue for the current session.  
- Note: it looks like MacVim will correctly perform syntax highlighting if `syntax on` or `syntax enable` is simply omitted, so the workaround is to remove `syntax on` from the .vimrc.

### Expected behaviour

All search matches should be highlighted, including the one under the cursor.

### Version of Vim and architecture

9.0.0065, GUI, arm64

### Environment

Monterey 12.5 for M2
Apple Terminal
xterm-256color
zsh 5.8.1 (x86_64-apple-darwin21.0)

### How MacVim was installed

Homebrew

### Vim configuration where issue is reproducable

https://gist.github.com/AbePralle/4083bbac7866782876250133ce831eb3

### Issue has been tested with given configuration

- [X] by running MacVim.app from GUI macOS interface
- [X] by running vim/gvim/etc installed by MacVim
- [ ] by running other versions of vim (e.g. /usr/bin/vim)

### Issue has been tested with no configuration

- [ ] by running `mvim --clean` (or `gvim`, supplied by MacVim distribution)
- [ ] by running `vim --clean` (in terminal, supplied by MacVim distribution)
- [ ] by running `vim --clean` (in terminal, other suppliers, e.g. /usr/bin/vim)

### Other conditions

- [ ] The both Homebrew packages "vim" and "macvim" are installed

**Comment by ychin** [comment]
I think this is a Vim bug, and from what I can see it was fixed in release 174 (9.0.472). Can you update MacVim and try again?

**Comment by AbePralle** [comment]
Yes, that fixes it. Thank you!
